lib/rucio/core/transfer.py

```diff
@@ -1096,7 +1096,7 @@ def prepare_transfers(
             logger(logging.WARNING, '%s: all available sources were filtered', rws)
             continue
 
-        update_dict = {
+        update_dict: Dict[Any, Any] = {
```

Comment:
How did this not raise an exception? `Dict` and `Any` is not imported under normal execution.

Reply:
This is scary

Reply:
I can fix it in #6277 if you want since this line is modified, but I am baffled.

Reply:
It would appear that this is working perfectly fine, though we can’t explain why. :man_shrugging:

Reply:
Would still be nice if you fix it in #6277

Reply:
From [PEP-526](https://peps.python.org/pep-0526/#runtime-effects-of-type-annotations):

> Annotating a local variable will cause the interpreter to treat it as a local, even if it was never assigned to. Annotations for local variables will not be evaluated:
> 
> ```
> def f():
>     x: NonexistentName  # No error.
> ```